x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -748,6 +748,88 @@ public void testOutOfRangeFilterPushdown() {
         }
     }
 
+    public void testOutOfRangeFilterForDoubleFieldsPushdown() {
+        var allTypeMappingAnalyzer = makeAnalyzer("mapping-all-types.json", new EnrichResolution());
+
+        String smallerThanFloat = String.valueOf(randomDoubleBetween(-Double.MAX_VALUE, Float.MIN_VALUE - 1d, true));
+        String largerThanFloat = String.valueOf(randomDoubleBetween(Float.MAX_VALUE + 1d, Double.MAX_VALUE, true));
+
+        List<OutOfRangeTestCase> cases = List.of(
+            new OutOfRangeTestCase("float", smallerThanFloat, largerThanFloat),
+            new OutOfRangeTestCase("half_float", smallerThanFloat, largerThanFloat)
+        );
+
+        final String LT = "<";
+        final String LTE = "<=";
+        final String GT = ">";
+        final String GTE = ">=";
+        final String EQ = "==";
+        final String NEQ = "!=";
+
+        for (OutOfRangeTestCase testCase : cases) {
+            List<String> rangePredicates = List.of(
+                LT + testCase.tooHigh,
+                LTE + testCase.tooHigh,
+                GT + testCase.tooLow,
+                GTE + testCase.tooLow,
+                LT + testCase.tooLow,
+                LTE + testCase.tooLow,
+                GT + testCase.tooHigh,
+                GTE + testCase.tooHigh
+            );
+
+            for (String rangePredicate : rangePredicates) {
+                String comparison = testCase.fieldName + rangePredicate;
+                var query = "from test | where " + comparison;
+
+                Source expectedSource = new Source(1, 18, comparison);
+
+                logger.info("Query: " + query);
+                EsQueryExec actualQueryExec = doTestOutOfRangeFilterPushdown(query, allTypeMappingAnalyzer);
+
+                assertThat(actualQueryExec.query(), is(instanceOf(SingleValueQuery.Builder.class)));
+                var actualLuceneQuery = (SingleValueQuery.Builder) actualQueryExec.query();
+                assertThat(actualLuceneQuery.field(), equalTo(testCase.fieldName));
+                assertThat(actualLuceneQuery.source(), equalTo(expectedSource));
+                assertTrue(actualLuceneQuery.next() instanceof RangeQueryBuilder);
+            }
+
+            for (var value : List.of(testCase.tooHigh, testCase.tooLow)) {
+                String comparison = testCase.fieldName + EQ + value;
+                var query = "from test | where " + comparison;
+
+                Source expectedSource = new Source(1, 18, comparison);
+
+                logger.info("Query: " + query);
+                EsQueryExec actualQueryExec = doTestOutOfRangeFilterPushdown(query, allTypeMappingAnalyzer);
+
+                var actualLuceneQuery = (SingleValueQuery.Builder) actualQueryExec.query();
+                assertThat(actualLuceneQuery.field(), equalTo(testCase.fieldName));
+                assertThat(actualLuceneQuery.source(), equalTo(expectedSource));
```

Comment:
All these are nearly identical in all loops, might be more readable if you declared a local reusable lambda?

Reply:
I appreciate the review 👍 - I will try to fold all these tests in the existing `testOutOfRangeFilterPushdown` since they are quite similar and many of the review comments could apply to `testOutOfRangeFilterPushdown` as well.

Reply:
Folding in the tests in the existing `testOutOfRangeFilterPushdown` is not looking any better - but I refactored the code a bit with your suggestions and it is not a bit more compact and hopefully more readable